src/library-authoring/component-info/ComponentInfo.tsx

```diff
@@ -39,6 +42,20 @@ const ComponentInfo = () => {
       category: getBlockType(usageKey),
     }, '*');
   };
+  const publishComponent = usePublishComponent(usageKey);
+  const { data: componentMetadata } = useLibraryBlockMetadata(usageKey);
+  // Only can be published when the component has been modified after the last published date.
+  const canPublish = (new Date(componentMetadata?.modified ?? 0)) > (new Date(componentMetadata?.lastPublished ?? 0));
```

Comment:
if we don't have data about the modified date we shouldn't be able to publish, and because of this we default the date to 0.